pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -334,6 +349,133 @@ public RepositoryData generate() {
         return repackage(new File(repoParentDir, "maven-repository"));
     }
 
+    public RepositoryData resolveAndRepackage() {
+        try {
+            log.info("Generating maven repository");
+            File sourceDir = createMavenGenerationDir();
+            log.info("Source dir " + sourceDir.getAbsolutePath());
+            boolean tempBuild = PigContext.get().isTempBuild();
+            final MavenArtifactResolver mvnResolver = MavenArtifactResolver.builder()
+                    .setUserSettings(new File(FileUtils.getConfiguredIndySettingsXmlPath(tempBuild)))
+                    .setLocalRepository(sourceDir.getAbsolutePath())
+                    .build();
+
+            Map<String, String> params = pigConfiguration.getFlow().getRepositoryGeneration().getParameters();
+            String bomConstraintGroupId = generationData.getBomGroupId();
+            String bomConstraintArtifactId = generationData.getBomArtifactId();
+            String bomConstraintVersion = getRedhatVersion(
+                    new org.jboss.da.model.rest.GAV(
+                            bomConstraintGroupId,
+                            bomConstraintArtifactId,
+                            generationData.getBomVersion()),
+                    tempBuild);
+
+            final Artifact bom = new DefaultArtifact(
+                    bomConstraintGroupId,
+                    bomConstraintArtifactId,
+                    null,
+                    "pom",
+                    bomConstraintVersion);
+            final List<Dependency> bomConstraints = mvnResolver.resolveDescriptor(bom).getManagedDependencies();
+            if (bomConstraints.isEmpty()) {
+                throw new IllegalStateException("Failed to resolve " + bom);
+            }
+            // Must point to a text file which contains list of format "groupId:artifactId:version"
+            String extensionsListUrl = params.get("extensionsListUrl");
+
+            List<Artifact> extensionRtArtifactList = parseExtensionsArtifactList(extensionsListUrl);
+            for (Artifact extensionRtArtifact : extensionRtArtifactList) {
+                // this will resolve all the runtime dependencies and as a consequence populate the local Maven repo
+                // specified in the user settings.xml
+                DependencyResult result = mvnResolver.resolveManagedDependencies(
+                        extensionRtArtifact, // runtime extension artifact
+                        Collections.emptyList(), // enforced direct dependencies, ignore this
+                        bomConstraints, // version constraints from the BOM
+                        Collections.emptyList(), // extra maven repos, ignore this
+                        "test",
+                        "provided" // dependency scopes that should be ignored
+                );
+
+            }
+            return repackage(sourceDir);
+        } catch (Exception bme) {
+            bme.printStackTrace();
+            return null;
```

Comment:
why is the exception swallowed here? Is there any case that bacon could continue after it?

Reply:
`repositoryGeneration functions` dont throw the exceptions. Exceptions are handled/thrown by the calling function. None of the repository Generation method throws exception

Reply:
that's not true, they throw `RuntimeException`s

Reply:
take a look e.g. at `RepoBuilder`

Reply:
Throwing Runtime exception now